File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i, j = 0, 0
        n = len(nums)
        cnt = 1
        while j < n:
            if nums[i] == nums[j]:
                j += 1
            else:
                i += 1
                nums[i], nums[j] = nums[j], nums[i]
                cnt += 1
                j += 1
        return cnt

File: Data Structures & Algorithms/majority-element-ii/submission-0.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int cnt1 = 0, cnt2 = 0, a = 0, b = 0;
        int n = nums.size();
        int chk = n/3;
        vector<int> answer;
        for(int i=0; i<n; i++){
            if(cnt1==0){
                a = nums[i];
                cnt1 = 1;
            }
            else if(nums[i] != a && cnt2==0){
                b = nums[i];
                cnt2 = 1;
            }
            else if(nums[i] == a){
                cnt1++;
            }
            else if(nums[i] == b){
                cnt2++;
            }
            else{
                cnt1--;
                cnt2--;
            }
        }
        cnt1=0, cnt2=0;
        for(int i=0; i<n; i++){
            if(nums[i] == a)    cnt1++;
            if(nums[i] == b)    cnt2++;
        }
        if(cnt1>chk)    answer.push_back(a);
        if(cnt2>chk)    answer.push_back(b);
        return answer;
    }
};